x-pack/legacy/plugins/ml/public/services/forecast_service.js

```diff
@@ -192,117 +193,112 @@ function getForecastData(
+      }
     });
+  });
 
 
 
-    // If an aggType object has been passed in, use it.
-    // Otherwise default to avg, min and max aggs for the
-    // forecast prediction, upper and lower
-    const forecastAggs = (aggType === undefined) ?
-      { avg: 'avg', max: 'max', min: 'min' } :
-      {
-        avg: aggType.avg,
-        max: aggType.max,
-        min: aggType.min
-      };
+  // If an aggType object has been passed in, use it.
+  // Otherwise default to avg, min and max aggs for the
+  // forecast prediction, upper and lower
+  const forecastAggs = (aggType === undefined) ?
+    { avg: 'avg', max: 'max', min: 'min' } :
+    {
+      avg: aggType.avg,
+      max: aggType.max,
+      min: aggType.min
+    };
 
-    ml.esSearch({
-      index: ML_RESULTS_INDEX_PATTERN,
-      size: 0,
-      body: {
-        query: {
-          bool: {
-            filter: filterCriteria
-          }
-        },
-        aggs: {
-          times: {
-            date_histogram: {
-              field: 'timestamp',
-              interval: interval,
-              min_doc_count: 1
+  return ml.esSearchRx({
+    index: ML_RESULTS_INDEX_PATTERN,
+    size: 0,
+    body: {
+      query: {
+        bool: {
+          filter: filterCriteria
+        }
+      },
+      aggs: {
+        times: {
+          date_histogram: {
+            field: 'timestamp',
+            interval: interval,
+            min_doc_count: 1
+          },
+          aggs: {
+            prediction: {
+              [forecastAggs.avg]: {
+                field: 'forecast_prediction'
+              }
+            },
+            forecastUpper: {
+              [forecastAggs.max]: {
+                field: 'forecast_upper'
+              }
             },
-            aggs: {
-              prediction: {
-                [forecastAggs.avg]: {
-                  field: 'forecast_prediction'
-                }
-              },
-              forecastUpper: {
-                [forecastAggs.max]: {
-                  field: 'forecast_upper'
-                }
-              },
-              forecastLower: {
-                [forecastAggs.min]: {
-                  field: 'forecast_lower'
-                }
+            forecastLower: {
+              [forecastAggs.min]: {
+                field: 'forecast_lower'
               }
             }
           }
         }
       }
-    })
-      .then((resp) => {
-        const aggregationsByTime = _.get(resp, ['aggregations', 'times', 'buckets'], []);
-        _.each(aggregationsByTime, (dataForTime) => {
-          const time = dataForTime.key;
-          obj.results[time] = {
-            prediction: _.get(dataForTime, ['prediction', 'value']),
-            forecastUpper: _.get(dataForTime, ['forecastUpper', 'value']),
-            forecastLower: _.get(dataForTime, ['forecastLower', 'value'])
-          };
-        });
-
-        resolve(obj);
-      })
-      .catch((resp) => {
-        reject(resp);
+    }
+  }).pipe(
```

Comment:
Does this need a `catchError()`? The Promise based version had `catch -> reject`. If I get it right the way it's written now we'll always return the `obj` with `{ success: true, ... }`.

Reply:
No, there is no need in `catchError`. In case of an error it will and up in `.catch` callback. We only need this operator if we want to handle an error and map to some other value so it will end up in `.subscribe` callback